stdlib/runtime/gc.gr

```diff
@@ -84,24 +85,37 @@ let rec decRef = (userPtr: WasmI32, ignoreZeros: Bool) => {
     // }
 
     if (WasmI32.eqz(refCount)) {
-      if (ignoreZeros) {
-        userPtr
-      } else {
+      if (!ignoreZeros) {
         throwDecRefError()
       }
     } else {
       let refCount = refCount - 1n
       setRefCount(userPtr, refCount)
 
       if (WasmI32.eqz(refCount)) {
-        decRefChildren(userPtr)
+        /*
+         * Note: We dangerously call free before decRefChildren to allow for
+         *       tco optimization. This works because `free` only merges the
+         *       empty blocks and does not actually zero out the memory.
```

Comment:
It's not dangerous if we know what we're doing 😉
```suggestion
         * Note: We call free before decRefChildren to allow for a tail call.
         * This is okay because no allocations occur while we traverse the
         * structure and free does not mangle the data.
```